client/src/lib/components/ShowMoreText/ShowMoreText.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+import ReadMore from '@fawazahmed/react-native-read-more';
```

Comment:
Could you explain the benefits of this lib? Couldn't we just do this with a character count and simple state in this component? The lib is a bit sloppy built (e.g. hard to find the source code) so I'm a bit hesitant at adding it.

Reply:
aha, I approached it similarly (mindset wise) using
```typescript
onTextLayout={({ nativeEvent: { lines } }) =>
    setShowMoreButton(lines.length > numberOfLines)
    ...
```

But it didn't work well also couldn't add the show more button by the end of the last line, would be interesting trying with character count as you mentioned

Reply:
on the example above I'm using `numberOfLines={5}` on the text to limit

Reply:
Expo seems to have a lib for this as well https://github.com/expo/react-native-read-more-text

Reply:
They seem to use an api to find out the height of the text component
https://github.com/expo/react-native-read-more-text/blob/3e63ca30bd0b7925893cf76bc015ab7e9b7a75a3/index.js#L103-L105

Reply:
Shall I copy their approach and do a component in our codebase? wdyt? @gewfy

Reply:
Thanks for doing some research here! I believe we did something similar, height based, in the old. Let me check :)

Reply:
Yeach exactly, in the old ap we did what you proposed ⬆️ 
```js
                onTextLayout={({ nativeEvent: { lines } }) => {
                  if (lines.length > NUMBER_OF_LINES && isNil(truncated)) {
                    setTruncated(true);
                  }
                }}
```

And then just absolute position a `Show more` at the right bottom of that component, or bellow.

Reply:
Since this seems more complex than it looks on the surface I leave it up to you. Fine to go with the lib you chose too :)